Scripts/Pickups/AirDashItem.cs

```diff
@@ -0,0 +1,72 @@
+using Godot;
+using System;
+
+public partial class AirDashItem : Area2D
+{
+    // States
+    public bool Collected = false;
+
```

Comment:
Refactor these three to a PickupArea2D class that extends area2D, basically just copy one of these, then add an abstract method called "ProcessPickup" or something

On the AirDashItem, DashItem, and DoubleJumpItem classes it'll just be Extending PickupArea2D and then defining our new "ProcessPickup" method which should handle the things like player.CanDash and the GD.Print about the power turning on.

In the Parent class' OnBodyEntered Method, call ProcessPickup() then the rest of OnBodyEntered, which in this case is setting the collected flag to true.

Reply:
Check out the most recent commit, I believe that's what you're talking about.